test/unit/logic/broadcaster.js

```diff
@@ -331,79 +316,285 @@ describe('Broadcaster', () => {
+					beforeEach(done => {
+						transactionStub.checkConfirmed.callsArgWith(
+							1,
+							'Checking if transction is confirmed error',
+							false
+						);
+						done();
+					});
+					it('should set an empty broadcaster.queue and skip the broadcast', done => {
+						filterQueue(() => {
+							expect(broadcaster.queue)
+								.to.be.an('Array')
+								.to.eql([]);
+							done();
+						});
+					});
+				});
+			});
 		});
 
-		it('should return false if transaction is not confirmed', done => {
-			modulesStub.transactions.transactionInPool.returns(false);
-			transactionStub.checkConfirmed.callsArgWith(1, true);
-			filterTransaction({ id: '123' }, (err, status) => {
-				expect(err).to.be.null;
-				expect(status).to.be.false;
+		describe('having many transaction and signatures broadcasts in queue', () => {
+			const broadcasts = [];
+			beforeEach(done => {
+				broadcaster.enqueue(params, {
+					api: 'postTransactions',
+					data: { transaction: { id: 1 } },
+					immediate: false,
+				});
+				broadcaster.enqueue(params, {
+					api: 'postTransactions',
+					data: { transaction: { id: 2 } },
+					immediate: false,
+				});
+				broadcaster.enqueue(params, {
+					api: 'postTransactions',
+					data: { transaction: { id: 3 } },
+					immediate: false,
+				});
+				broadcasts.push(
+					Object.assign(
+						{},
+						{ params },
+						{
+							options: {
+								api: 'postTransactions',
+								data: { transaction: { id: 1 } },
+								immediate: false,
+							},
+						}
+					)
+				);
+				broadcasts.push(
+					Object.assign(
+						{},
+						{ params },
+						{
+							options: {
+								api: 'postTransactions',
+								data: { transaction: { id: 2 } },
+								immediate: false,
+							},
+						}
+					)
+				);
+				broadcasts.push(
+					Object.assign(
+						{},
+						{ params },
+						{
+							options: {
+								api: 'postTransactions',
+								data: { transaction: { id: 3 } },
+								immediate: false,
+							},
+						}
+					)
+				);
+				broadcaster.enqueue(params, {
+					api: 'postSignatures',
+					data: { signature: { transactionId: 1 } },
+					immediate: false,
+				});
+				broadcaster.enqueue(params, {
+					api: 'postSignatures',
+					data: { signature: { transactionId: 2 } },
+					immediate: false,
+				});
+				broadcasts.push(
+					Object.assign(
+						{},
+						{ params },
+						{
+							options: {
+								api: 'postSignatures',
+								data: { signature: { transactionId: 1 } },
+								immediate: false,
+							},
+						}
+					)
+				);
+				broadcasts.push(
+					Object.assign(
+						{},
+						{ params },
+						{
+							options: {
+								api: 'postSignatures',
+								data: { signature: { transactionId: 2 } },
+								immediate: false,
+							},
+						}
+					)
+				);
 				done();
 			});
+
+			describe('when all of them exist in transaction pool', () => {
+				beforeEach(done => {
```

Comment:
Do we really need to aisle this line inside a different beforeEach statement? Why?

Reply:
The describe description corresponds with a context that is being enforced by `beforeEach` statements. Like here: 
```
describe('having many transaction and signatures broadcasts in queue')
   beforeEach is setting many broadcasts to queue
   describe('when all of them exist in transaction pool')
      beforeEach is making transaction pool to return true always
```
The same context for "having many transactions and signatures broadcasts in queue'" might be extended by additional contexts like "when some of them are confirmed". I didn't go so far.